test/relations.test.js

```diff
@@ -605,6 +605,864 @@ describe('relations', function() {
+                  if (err) return done(err);
+                  verify(physician, patient.id);
+                });
+              });
+            });
+          });
+          function verify(physician, patientId) {
+            physician.patients({where: {id: {inq: [patientId]}}}, function(err, ch) {
+              if (err) return done(err);
+              should.exist(ch);
+              ch.should.have.lengthOf(1);
+              ch[0].age.should.eql(5);
+              done();
+            });
+          }
+        });
+
+        it('should find scoped record with promises based on related model properties', function(done) {
+          let id;
+          Physician.create()
+            .then(function(physician) {
+              return physician.patients.create({name: 'a'})
+                .then(function(ch) {
+                  id = ch.id;
+                  return physician.patients.create({name: 'z'});
+                })
+                .then(function() {
+                  return physician.patients.create({name: 'c'});
+                })
+                .then(function() {
+                  return verify(physician);
+                });
+            }).catch(done);
+
+          function verify(physician) {
+            return physician.patients({
+              where: {
+                name: 'a',
+              },
+            }, function(err, patients) {
+              if (err) return done(err);
+              should.exist(patients);
+              patients.should.have.lengthOf(1);
+              patients[0].name.should.equal('a');
+              done();
+            });
+          }
+        });
+
+        it('should find scoped record with promises based on related model' +
+          ' properties with empty results', function(done) {
+          let id;
+          Physician.create()
+            .then(function(physician) {
+              return physician.patients.create({name: 'a'})
+                .then(function(ch) {
+                  id = ch.id;
+                  return physician.patients.create({name: 'z'});
+                })
+                .then(function() {
+                  return physician.patients.create({name: 'c'});
+                })
+                .then(function() {
+                  return verify(physician);
+                });
+            }).catch(done);
+
+          function verify(physician) {
+            return physician.patients({
+              where: {
+                id: 'bar',
+              },
+            }, function(err, patients) {
+              if (err) return done(err);
+              should.exist(patients);
+              patients.should.have.lengthOf(0);
+              done();
+            });
+          }
+        });
+
+        it('should find scoped record with promises based on related model id', function(done) {
+          let id;
+          Physician.create()
+            .then(function(physician) {
```

Comment:
Similarly here. Please use `await` instead of Promise/`.then()` style.

```ts
const physician = await Physician.create();
```

Reply:
(The same comment applies to all other places where you are using `.then()` or `.catch()`.)

Reply:
Fixed